### ISSUE TITLE
Webrender asserts when rendering very long unbroken text runs

### ISSUE THREAD

**Issue by jdm** [servo/servo#22593]
When rendering a text run generated with:
```html
<body><div>
<script>
wellCompressableData = '';
 for (var i = 0; i < 500; i++) { wellCompressableData += 'foofoofoofoofoofoofoo'; }
document.body.firstChild.innerText = wellCompressableData;
</script>
```
Webrender yields this assertion:
```
0:07.03 pid:16661 assertion failed: request.current_used_block_num() <= MAX_VERTEX_TEXTURE_WIDTH (thread WRRenderBackend#0, at /Users/jdm/.cargo/git/checkouts/webrender-c3596abe1cf4f320/323b7ea/webrender/src/prim_store.rs:1094)
 0:10.14 pid:16661 stack backtrace:
 0:10.14 pid:16661    0:        0x113aa6b94 - __ZN9backtrace9backtrace5trace17hefbd7c3b17ad6e6eE
 0:10.14 pid:16661                         at /Users/jdm/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.9/src/backtrace/mod.rs:42
 0:10.14 pid:16661    1:        0x113a9fc3c - __ZN9backtrace7capture9Backtrace14new_unresolved17hfa7d39a5ef5fb84fE
 0:10.14 pid:16661                         at /Users/jdm/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.9/src/capture.rs:88
 0:10.14 pid:16661    2:        0x113a9fb9e - __ZN9backtrace7capture9Backtrace3new17hb34b0ddcba79179eE
 0:10.14 pid:16661                         at /Users/jdm/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.9/src/capture.rs:63
 0:10.14 pid:16661    3:        0x10db44590 - __ZN5servo4main28_$u7b$$u7b$closure$u7d$$u7d$17h7f3fe551d03f80b0E
 0:10.14 pid:16661                         at ports/servo/non_android_main.rs:111
 0:10.14 pid:16661    4:        0x11586a6f8 - __ZN3std9panicking20rust_panic_with_hook17h14490a1041ee4235E
 0:10.14 pid:16661                         at libstd/panicking.rs:480
 0:10.14 pid:16661    5:        0x1154cda87 - __ZN3std9panicking11begin_panic17h7ad720de9aa9d9e6E
 0:10.14 pid:16661                         at libstd/panicking.rs:410
 0:10.14 pid:16661    6:        0x112c50f64 - __ZN9webrender10prim_store21PrimitiveTemplateKind21write_prim_gpu_blocks17h2948eab38215508dE
 0:10.14 pid:16661                         at /Users/jdm/.cargo/git/checkouts/webrender-c3596abe1cf4f320/323b7ea/webrender/src/prim_store.rs:1094
 0:10.14 pid:16661    7:        0x112c51813 - __ZN9webrender10prim_store17PrimitiveTemplate6update17ha7a9aaedf59db55fE
 0:10.14 pid:16661                         at /Users/jdm/.cargo/git/checkouts/webrender-c3596abe1cf4f320/323b7ea/webrender/src/prim_store.rs:1225
 0:10.14 pid:16661    8:        0x112c58f23 - __ZN9webrender10prim_store14PrimitiveStore32prepare_interned_prim_for_render17h3587ac7845c20341E
 0:10.14 pid:16661                         at /Users/jdm/.cargo/git/checkouts/webrender-c3596abe1cf4f320/323b7ea/webrender/src/prim_store.rs:3215
 0:10.14 pid:16661    9:        0x112c58290 - __ZN9webrender10prim_store14PrimitiveStore23prepare_prim_for_render17h318f878f7b6074f9E
 0:10.14 pid:16661                         at /Users/jdm/.cargo/git/checkouts/webrender-c3596abe1cf4f320/323b7ea/webrender/src/prim_store.rs:3093
 0:10.14 pid:16661   10:        0x112c58dc8 - __ZN9webrender10prim_store14PrimitiveStore18prepare_primitives17hb1b1a047ed708f2fE
 0:10.14 pid:16661                         at /Users/jdm/.cargo/git/checkouts/webrender-c3596abe1cf4f320/323b7ea/webrender/src/prim_store.rs:3181
 0:10.14 pid:16661   11:        0x112c5736c - __ZN9webrender10prim_store14PrimitiveStore23prepare_prim_for_render17h318f878f7b6074f9E
 0:10.14 pid:16661                         at /Users/jdm/.cargo/git/checkouts/webrender-c3596abe1cf4f320/323b7ea/webrender/src/prim_store.rs:2848
 0:10.15 pid:16661   12:        0x112c58dc8 - __ZN9webrender10prim_store14PrimitiveStore18prepare_primitives17hb1b1a047ed708f2fE
 0:10.15 pid:16661                         at /Users/jdm/.cargo/git/checkouts/webrender-c3596abe1cf4f320/323b7ea/webrender/src/prim_store.rs:3181
 0:10.15 pid:16661   13:        0x112cd999d - __ZN9webrender13frame_builder12FrameBuilder40build_layer_screen_rects_and_cull_layers17h99d1d916ec5e660aE
 0:10.15 pid:16661                         at /Users/jdm/.cargo/git/checkouts/webrender-c3596abe1cf4f320/323b7ea/webrender/src/frame_builder.rs:273
 0:10.15 pid:16661   14:        0x112cda1fb - __ZN9webrender13frame_builder12FrameBuilder5build17h5da3cef0a315190fE
 0:10.15 pid:16661                         at /Users/jdm/.cargo/git/checkouts/webrender-c3596abe1cf4f320/323b7ea/webrender/src/frame_builder.rs:359
 0:10.15 pid:16661   15:        0x112d7ed77 - __ZN9webrender14render_backend8Document11build_frame17hddc4f4daf62658c7E
 0:10.15 pid:16661                         at /Users/jdm/.cargo/git/checkouts/webrender-c3596abe1cf4f320/323b7ea/webrender/src/render_backend.rs:418
 0:10.15 pid:16661   16:        0x112d85a20 - __ZN9webrender14render_backend13RenderBackend15update_document17h2e57396c47082177E
 0:10.15 pid:16661                         at /Users/jdm/.cargo/git/checkouts/webrender-c3596abe1cf4f320/323b7ea/webrender/src/render_backend.rs:1219
 0:10.15 pid:16661   17:        0x112d812a8 - __ZN9webrender14render_backend13RenderBackend3run17ha020e7411e8b817dE
 0:10.15 pid:16661                         at /Users/jdm/.cargo/git/checkouts/webrender-c3596abe1cf4f320/323b7ea/webrender/src/render_backend.rs:773
 0:10.15 pid:16661   18:        0x112d958ee - __ZN9webrender8renderer8Renderer3new28_$u7b$$u7b$closure$u7d$$u7d$17hd20a1a49d81f9b8cE
 0:10.15 pid:16661                         at /Users/jdm/.cargo/git/checkouts/webrender-c3596abe1cf4f320/323b7ea/webrender/src/renderer.rs:1952
 0:10.15 pid:16661   19:        0x11306e224 - __ZN3std10sys_common9backtrace28__rust_begin_short_backtrace17h2fa3025b78fc0987E
 0:10.15 pid:16661                         at libstd/sys_common/backtrace.rs:136
 0:10.15 pid:16661   20:        0x113070ac4 - __ZN3std6thread7Builder5spawn28_$u7b$$u7b$closure$u7d$$u7d$28_$u7b$$u7b$closure$u7d$$u7d$17hb88311c41eb6f05aE
 0:10.15 pid:16661                         at libstd/thread/mod.rs:409
 0:10.15 pid:16661   21:        0x112d42804 - __ZN101_$LT$std..panic..AssertUnwindSafe$LT$F$GT$$u20$as$u20$core..ops..function..FnOnce$LT$$LP$$RP$$GT$$GT$9call_once17h882325bcceab2c62E
 0:10.15 pid:16661                         at libstd/panic.rs:313
 0:10.15 pid:16661   22:        0x112d45164 - __ZN3std9panicking3try7do_call17h8c36c0b0338b1c65E
 0:10.15 pid:16661                         at libstd/panicking.rs:310
 0:10.15 pid:16661   23:        0x115876c6e - ___rust_maybe_catch_panic
 0:10.15 pid:16661                         at libpanic_unwind/lib.rs:102
 0:10.15 pid:16661   24:        0x112d44953 - __ZN3std9panicking3try17hb78bdab6f6286dc1E
 0:10.15 pid:16661                         at libstd/panicking.rs:289
 0:10.15 pid:16661   25:        0x112d43d84 - __ZN3std5panic12catch_unwind17hcc795676d36bd5ccE
 0:10.15 pid:16661                         at libstd/panic.rs:392
 0:10.15 pid:16661   26:        0x11307085e - __ZN3std6thread7Builder5spawn28_$u7b$$u7b$closure$u7d$$u7d$17h914ad0f5fe973165E
 0:10.15 pid:16661                         at libstd/thread/mod.rs:408
 0:10.15 pid:16661   27:        0x1130be923 - __ZN50_$LT$F$u20$as$u20$alloc..boxed..FnBox$LT$A$GT$$GT$8call_box17h26b8d9da9b8e3f2cE
 0:10.15 pid:16661                         at liballoc/boxed.rs:672
 0:10.15 pid:16661   28:        0x11584dc0b - __ZN3std3sys4unix6thread6Thread3new12thread_start17hba53abd905cebc29E
 0:10.15 pid:16661                         at libstd/sys_common/thread.rs:24
 0:10.15 pid:16661   29:     0x7fffae9ec93a - __pthread_body
 0:10.15 pid:16661   30:     0x7fffae9ec886 - __pthread_start
 0:10.19 pid:16661 assertion failed: block_count <= self.max_block_count (thread WRRenderBackend#0, at /Users/jdm/.cargo/git/checkouts/webrender-c3596abe1cf4f320/323b7ea/webrender/src/gpu_cache.rs:524)
 0:10.21 pid:16661 stack backtrace:
 0:10.21 pid:16661    0:        0x113aa6b94 - __ZN9backtrace9backtrace5trace17hefbd7c3b17ad6e6eE
 0:10.21 pid:16661                         at /Users/jdm/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.9/src/backtrace/mod.rs:42
 0:10.21 pid:16661    1:        0x113a9fc3c - __ZN9backtrace7capture9Backtrace14new_unresolved17hfa7d39a5ef5fb84fE
 0:10.21 pid:16661                         at /Users/jdm/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.9/src/capture.rs:88
 0:10.21 pid:16661    2:        0x113a9fb9e - __ZN9backtrace7capture9Backtrace3new17hb34b0ddcba79179eE
 0:10.21 pid:16661                         at /Users/jdm/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.9/src/capture.rs:63
 0:10.21 pid:16661    3:        0x10db44590 - __ZN5servo4main28_$u7b$$u7b$closure$u7d$$u7d$17h7f3fe551d03f80b0E
 0:10.21 pid:16661                         at ports/servo/non_android_main.rs:111
 0:10.21 pid:16661    4:        0x11586a6f8 - __ZN3std9panicking20rust_panic_with_hook17h14490a1041ee4235E
 0:10.21 pid:16661                         at libstd/panicking.rs:480
 0:10.21 pid:16661    5:        0x1154cda87 - __ZN3std9panicking11begin_panic17h7ad720de9aa9d9e6E
 0:10.21 pid:16661                         at libstd/panicking.rs:410
 0:10.22 pid:16661    6:        0x1130fbc89 - __ZN92_$LT$webrender..gpu_cache..GpuDataRequest$LT$$u27$a$GT$$u20$as$u20$core..ops..drop..Drop$GT$4drop17hba37091f5dd26618E
 0:10.22 pid:16661                         at /Users/jdm/.cargo/git/checkouts/webrender-c3596abe1cf4f320/323b7ea/webrender/src/gpu_cache.rs:524
 0:10.22 pid:16661    7:        0x11309e7c4 - __ZN4core3ptr13drop_in_place17hb858f60311873c1cE
 0:10.22 pid:16661                         at libcore/ptr.rs:194
 0:10.22 pid:16661    8:        0x112c51886 - __ZN9webrender10prim_store17PrimitiveTemplate6update17ha7a9aaedf59db55fE
 0:10.22 pid:16661                         at /Users/jdm/.cargo/git/checkouts/webrender-c3596abe1cf4f320/323b7ea/webrender/src/prim_store.rs:1227
 0:10.22 pid:16661    9:        0x112c58f23 - __ZN9webrender10prim_store14PrimitiveStore32prepare_interned_prim_for_render17h3587ac7845c20341E
 0:10.22 pid:16661                         at /Users/jdm/.cargo/git/checkouts/webrender-c3596abe1cf4f320/323b7ea/webrender/src/prim_store.rs:3215
 0:10.22 pid:16661   10:        0x112c58290 - __ZN9webrender10prim_store14PrimitiveStore23prepare_prim_for_render17h318f878f7b6074f9E
 0:10.22 pid:16661                         at /Users/jdm/.cargo/git/checkouts/webrender-c3596abe1cf4f320/323b7ea/webrender/src/prim_store.rs:3093
 0:10.22 pid:16661   11:        0x112c58dc8 - __ZN9webrender10prim_store14PrimitiveStore18prepare_primitives17hb1b1a047ed708f2fE
 0:10.22 pid:16661                         at /Users/jdm/.cargo/git/checkouts/webrender-c3596abe1cf4f320/323b7ea/webrender/src/prim_store.rs:3181
 0:10.22 pid:16661   12:        0x112c5736c - __ZN9webrender10prim_store14PrimitiveStore23prepare_prim_for_render17h318f878f7b6074f9E
 0:10.22 pid:16661                         at /Users/jdm/.cargo/git/checkouts/webrender-c3596abe1cf4f320/323b7ea/webrender/src/prim_store.rs:2848
 0:10.22 pid:16661   13:        0x112c58dc8 - __ZN9webrender10prim_store14PrimitiveStore18prepare_primitives17hb1b1a047ed708f2fE
 0:10.22 pid:16661                         at /Users/jdm/.cargo/git/checkouts/webrender-c3596abe1cf4f320/323b7ea/webrender/src/prim_store.rs:3181
 0:10.22 pid:16661   14:        0x112cd999d - __ZN9webrender13frame_builder12FrameBuilder40build_layer_screen_rects_and_cull_layers17h99d1d916ec5e660aE
 0:10.22 pid:16661                         at /Users/jdm/.cargo/git/checkouts/webrender-c3596abe1cf4f320/323b7ea/webrender/src/frame_builder.rs:273
 0:10.22 pid:16661   15:        0x112cda1fb - __ZN9webrender13frame_builder12FrameBuilder5build17h5da3cef0a315190fE
 0:10.22 pid:16661                         at /Users/jdm/.cargo/git/checkouts/webrender-c3596abe1cf4f320/323b7ea/webrender/src/frame_builder.rs:359
 0:10.22 pid:16661   16:        0x112d7ed77 - __ZN9webrender14render_backend8Document11build_frame17hddc4f4daf62658c7E
 0:10.22 pid:16661                         at /Users/jdm/.cargo/git/checkouts/webrender-c3596abe1cf4f320/323b7ea/webrender/src/render_backend.rs:418
 0:10.22 pid:16661   17:        0x112d85a20 - __ZN9webrender14render_backend13RenderBackend15update_document17h2e57396c47082177E
 0:10.22 pid:16661                         at /Users/jdm/.cargo/git/checkouts/webrender-c3596abe1cf4f320/323b7ea/webrender/src/render_backend.rs:1219
 0:10.22 pid:16661   18:        0x112d812a8 - __ZN9webrender14render_backend13RenderBackend3run17ha020e7411e8b817dE
 0:10.22 pid:16661                         at /Users/jdm/.cargo/git/checkouts/webrender-c3596abe1cf4f320/323b7ea/webrender/src/render_backend.rs:773
 0:10.22 pid:16661   19:        0x112d958ee - __ZN9webrender8renderer8Renderer3new28_$u7b$$u7b$closure$u7d$$u7d$17hd20a1a49d81f9b8cE
 0:10.22 pid:16661                         at /Users/jdm/.cargo/git/checkouts/webrender-c3596abe1cf4f320/323b7ea/webrender/src/renderer.rs:1952
 0:10.22 pid:16661   20:        0x11306e224 - __ZN3std10sys_common9backtrace28__rust_begin_short_backtrace17h2fa3025b78fc0987E
 0:10.22 pid:16661                         at libstd/sys_common/backtrace.rs:136
 0:10.22 pid:16661   21:        0x113070ac4 - __ZN3std6thread7Builder5spawn28_$u7b$$u7b$closure$u7d$$u7d$28_$u7b$$u7b$closure$u7d$$u7d$17hb88311c41eb6f05aE
 0:10.22 pid:16661                         at libstd/thread/mod.rs:409
 0:10.22 pid:16661   22:        0x112d42804 - __ZN101_$LT$std..panic..AssertUnwindSafe$LT$F$GT$$u20$as$u20$core..ops..function..FnOnce$LT$$LP$$RP$$GT$$GT$9call_once17h882325bcceab2c62E
 0:10.22 pid:16661                         at libstd/panic.rs:313
 0:10.22 pid:16661   23:        0x112d45164 - __ZN3std9panicking3try7do_call17h8c36c0b0338b1c65E
 0:10.22 pid:16661                         at libstd/panicking.rs:310
 0:10.22 pid:16661   24:        0x115876c6e - ___rust_maybe_catch_panic
 0:10.22 pid:16661                         at libpanic_unwind/lib.rs:102
 0:10.22 pid:16661   25:        0x112d44953 - __ZN3std9panicking3try17hb78bdab6f6286dc1E
 0:10.22 pid:16661                         at libstd/panicking.rs:289
 0:10.22 pid:16661   26:        0x112d43d84 - __ZN3std5panic12catch_unwind17hcc795676d36bd5ccE
 0:10.22 pid:16661                         at libstd/panic.rs:392
 0:10.22 pid:16661   27:        0x11307085e - __ZN3std6thread7Builder5spawn28_$u7b$$u7b$closure$u7d$$u7d$17h914ad0f5fe973165E
 0:10.22 pid:16661                         at libstd/thread/mod.rs:408
 0:10.22 pid:16661   28:        0x1130be923 - __ZN50_$LT$F$u20$as$u20$alloc..boxed..FnBox$LT$A$GT$$GT$8call_box17h26b8d9da9b8e3f2cE
 0:10.22 pid:16661                         at liballoc/boxed.rs:672
 0:10.22 pid:16661   29:        0x11584dc0b - __ZN3std3sys4unix6thread6Thread3new12thread_start17hba53abd905cebc29E
 0:10.22 pid:16661                         at libstd/sys_common/thread.rs:24
 0:10.22 pid:16661   30:     0x7fffae9ec93a - __pthread_body
 0:10.22 pid:16661   31:     0x7fffae9ec886 - __pthread_start
```

**Comment by jdm** [comment]
This affects a bunch of tests under web-platform-tests/referrer-policy as well.

**Comment by MeFisto94** [comment]
Actually this is a duplicate of #17230, because servo fails to tile the text elements, probably it splits at space characters or something.

**Comment by jdm** [comment]
Fixed by #25289
